developer_tests/utilities/parse_args_test.f90

```diff
@@ -1,90 +1,198 @@
 
-   print *, 'input line: "'//trim(nextline)//'"'
-   print *, 'got ', nargs, ' arguments from the input line'
-   do i=1, nargs
-      print *, 'name  ', i, 'is: ', '"'//trim(names(i))//'"'
-      print *, 'value ', i, 'is: ', '"'//trim(vals(i))//'"'
-   enddo
-   print *, 'continue line = ', continue_line
+testnum = 'test 5'
+call resetvals()
+nextline = '"quoted string"'
+call get_args_from_string(nextline, nargs, args)
+
+call ok((nargs == 1),                 trim(testnum)//', nargs')
+call ok((args(1) == 'quoted string'), trim(testnum)//', arg 1')
+call ok((args(2) == ''),              trim(testnum)//', arg 2')
+
+
+testnum = 'test 6'
+call resetvals()
+nextline = '  "quoted string"  "second string  with blanks"  '
+call get_args_from_string(nextline, nargs, args)
+
+call ok((nargs == 2),                              trim(testnum)//', nargs')
+call ok((args(1) == 'quoted string'),              trim(testnum)//', arg 1')
+call ok((args(2) == 'second string  with blanks'), trim(testnum)//', arg 2')
+call ok((args(3) == ''),                           trim(testnum)//', arg 3')
+
+
+testnum = 'test 7'
+call resetvals()
+nextline = '"quoted" "string"'
+call get_args_from_string(nextline, nargs, args)
+
+call ok((nargs == 2),           trim(testnum)//', nargs')
+call ok((args(1) == 'quoted'),  trim(testnum)//', arg 1')
+call ok((args(2) == 'string'),  trim(testnum)//', arg 1')
+call ok((args(3) == ''),        trim(testnum)//', arg 2')
+
+
+testnum = 'test 8'
+call resetvals()
+nextline = 'string'//BACKSLASH//'abc'
+call get_args_from_string(nextline, nargs, args)
+ 
+call ok((nargs == 1),             trim(testnum)//', nargs')
+call ok((args(1) == 'stringabc'), trim(testnum)//', arg 1')
+call ok((args(2) == ''),          trim(testnum)//', arg 2')
+
+
+testnum = 'test 9'
+call resetvals()
+nextline = 'directory'//BACKSLASH//BACKSLASH//'abc'
+call get_args_from_string(nextline, nargs, args)
+
+call ok((nargs == 1),                               trim(testnum)//', nargs')
+call ok((args(1) == 'directory'//BACKSLASH//'abc'), trim(testnum)//', arg 1')
+call ok((args(2) == ''),                            trim(testnum)//', arg 2')
+
+
+! start of name/value pair tests
+
+testnum = 'test 10'
+call resetvals()
+nextline = 'a=1 b=2 c=3'
+call get_name_val_pairs_from_string(nextline, nargs, names, vals, continue_line)
+
+call ok((nargs == 3),           trim(testnum)//', nargs')
+call ok((names(1) == 'a'),      trim(testnum)//', name 1')
+call ok((vals(1)  == '1'),      trim(testnum)//', val 1')
+call ok((names(2) == 'b'),      trim(testnum)//', name 2')
+call ok((vals(2)  == '2'),      trim(testnum)//', val 2')
+call ok((names(3) == 'c'),      trim(testnum)//', name 3')
+call ok((vals(3)  == '3'),      trim(testnum)//', val 3')
+call ok((names(4) == ''),       trim(testnum)//', name 4')
+call ok((vals(4)  == ''),       trim(testnum)//', val 4')
+call ok((.not. continue_line),  trim(testnum)//', continue_line')
+
+
+testnum = 'test 10'
```

Comment:
Should this be test 11 since it is a new nextline?  Or is it 10 because it is testing name/value pairs?
```suggestion
testnum = 'test 11'
```

Reply:
fixed - cut n paste error